### PR TITLE
convert partition table tests to pytest

### PR DESCRIPTION
**Pull request by peterdragun**
Test results:
```
pytest -v tests/foreman/*/test_partitiontable.py 
======================================================= test session starts ========================================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 -- /home/pdragun/.virtualenvs/robottelo_master/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pdragun/Documents/robottelo
plugins: forked-1.0.2, xdist-1.29.0, mock-1.10.4, cov-2.7.1, services-1.3.1
collecting 69 items                                                                                                                2020-08-18 09:33:27 - conftest - DEBUG - Collected 69 test cases
collected 69 items                                                                                                                 

tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[0] PASSED         [  1%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[1] PASSED         [  2%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[2] PASSED         [  4%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[3] PASSED         [  5%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[4] PASSED         [  7%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[5] PASSED         [  8%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[6] PASSED         [ 10%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[0] PASSED                         [ 11%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[1] PASSED                         [ 13%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[2] PASSED                         [ 14%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[3] PASSED                         [ 15%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[4] PASSED                         [ 17%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[5] PASSED                         [ 18%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[6] PASSED                         [ 20%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[0] PASSED              [ 21%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[1] PASSED              [ 23%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[2] PASSED              [ 24%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[3] PASSED              [ 26%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[4] PASSED              [ 27%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[5] PASSED              [ 28%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_layout[6] PASSED              [ 30%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_with_layout_length PASSED                 [ 31%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_update_with_os PASSED                     [ 33%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_positive_create_search_with_org PASSED                    [ 34%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[0] PASSED               [ 36%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[1] PASSED               [ 37%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[2] PASSED               [ 39%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[3] PASSED               [ 40%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[4] PASSED               [ 42%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[5] PASSED               [ 43%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[6] PASSED               [ 44%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[7] PASSED               [ 46%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[8] PASSED               [ 47%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_invalid_name[9] PASSED               [ 49%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_empty_layout[0] PASSED               [ 50%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_empty_layout[1] PASSED               [ 52%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_create_with_empty_layout[2] PASSED               [ 53%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[0] PASSED                            [ 55%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[1] PASSED                            [ 56%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[2] PASSED                            [ 57%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[3] PASSED                            [ 59%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[4] PASSED                            [ 60%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[5] PASSED                            [ 62%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[6] PASSED                            [ 63%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[7] PASSED                            [ 65%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[8] PASSED                            [ 66%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_name[9] PASSED                            [ 68%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_layout[0] PASSED                          [ 69%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_layout[1] PASSED                          [ 71%]
tests/foreman/api/test_partitiontable.py::TestPartitionTable::test_negative_update_layout[2] PASSED                          [ 72%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[0] PASSED         [ 73%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[1] PASSED         [ 75%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[2] PASSED         [ 76%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[3] PASSED         [ 78%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[4] PASSED         [ 79%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[5] PASSED         [ 81%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_one_character_name[6] PASSED         [ 82%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[0] PASSED                         [ 84%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[1] PASSED                         [ 85%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[2] PASSED                         [ 86%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[3] PASSED                         [ 88%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[4] PASSED                         [ 89%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[5] PASSED                         [ 91%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_crud_with_name[6] PASSED                         [ 92%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_content PASSED                       [ 94%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_create_with_content_length PASSED                [ 95%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_delete_by_id PASSED                              [ 97%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_add_remove_os_by_id PASSED                       [ 98%]
tests/foreman/cli/test_partitiontable.py::TestPartitionTable::test_positive_add_remove_os_by_name PASSED                     [100%]
============================================= 69 passed, 7 warnings in 557.07 seconds ==============================================
```

### CODE REVIEW

**Comment by peterdragun** [comment]
PR was updated to use `:parametrized: yes` in docstring for correct uploading of all parametrized tests to Polarion.

File: tests/foreman/api/test_partitiontable.py [hunk 1]
```diff
@@ -99,26 +125,31 @@ def test_positive_create_with_layout_length(self):
...
-        :expectedresults: Partition table created successfully and has correct
+        :expectedresults: Partition table created and updated successfully and has correct
             operating system
 
         """
         os_family = OPERATING_SYSTEMS[randint(0, 8)]
         ptable = entities.PartitionTable(os_family=os_family).create()
-        self.assertEqual(ptable.os_family, os_family)
+        assert ptable.os_family == os_family
+        new_os_family = OPERATING_SYSTEMS[randint(0, 8)]
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_partitiontable.py:144]:
```suggestion
        new_os_family = random.choice(OPERATING_SYSTEMS)
```

**Review by mirekdlugosz — APPROVED** [review]
ACK with suggestion.

If you disagree with that change, we can merge as-is. But it would be better to make that change now and then merge, than to merge and create another PR for this simple change.

File: tests/foreman/api/test_partitiontable.py (hunk 1 above)
Comment (peterdragun) [reply tests/foreman/api/test_partitiontable.py]: thanks for suggestion, updated

**Pull request merged**